### ISSUE TITLE
Update conflict on enrichment

### ISSUE THREAD

**Issue by Xophmeister** [wtsi-hgi/cookie-monster#11]
The Cookie Jar's CouchDB is failing with an update conflict error when the `queue/set_processing` update handler (others?) is called.

Currently investigating causes; most likely due to near-simultaneous updates on the same document without guaranteed atomicity.

**Comment by colin-nolan** [comment]
Failure occurs on call to `BiscuitTin.mark_as_complete` following the processing of a cookie (with no termination, no more enrichments possible).

The error only occurs when there is more than one cookie processor.

Traceback:
```
Traceback (most recent call last):
  File "/usr/lib/python3.5/threading.py", line 914, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.5/threading.py", line 862, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.5/dist-packages/cookiemonster/processor/basic_processing.py", line 39, in process
    on_complete(terminate, notifications)
  File "/usr/local/lib/python3.5/dist-packages/cookiemonster/processor/basic_processing.py", line 82, in on_complete
    self.on_cookie_processed(cookie, rules_matched, notifications)
  File "/usr/local/lib/python3.5/dist-packages/cookiemonster/processor/basic_processing.py", line 113, in on_cookie_processed
    self._cookie_jar.mark_as_complete(cookie.path)
  File "/usr/local/lib/python3.5/dist-packages/cookiemonster/cookiejar/biscuit_tin.py", line 89, in mark_as_complete
    self._queue.mark_finished(path)
  File "/usr/local/lib/python3.5/dist-packages/cookiemonster/cookiejar/_dbi.py", line 493, in mark_finished
    self.upsert('queue', 'set_processing', doc_id, processing = False)
  File "/usr/local/lib/python3.5/dist-packages/cookiemonster/cookiejar/_dbi.py", line 296, in upsert
    res_headers, res_body = self._db.update_doc(update_name, key, **options)
  File "/usr/local/lib/python3.5/dist-packages/couchdb/client.py", line 941, in update_doc
    _, headers, body = func(**options)
  File "/usr/local/lib/python3.5/dist-packages/couchdb/http.py", line 504, in put
    return self._request('PUT', path, body=body, headers=headers, **params)
  File "/usr/local/lib/python3.5/dist-packages/couchdb/http.py", line 529, in _request
    credentials=self.credentials)
  File "/usr/local/lib/python3.5/dist-packages/couchdb/http.py", line 379, in request
    raise ResourceConflict(error)
couchdb.http.ResourceConflict: ('conflict', 'Document update conflict.')
```

**Comment by Xophmeister** [comment]
> Hopefully this resolves #11

Confirmed :smile:
